clientlibrary/checkpoint/dynamodb-checkpointer.go

```diff
@@ -119,6 +119,17 @@ func (checkpointer *DynamoCheckpoint) GetLease(shard *par.ShardStatus, newAssign
 		return err
 	}
 
+	var claimRequest string
+	if checkpointer.kclConfig.EnableLeaseStealing {
+		if currentCheckpointClaimRequest, ok := currentCheckpoint[CLAIM_REQUEST_KEY]; ok && currentCheckpointClaimRequest.S != nil {
+			claimRequest = *currentCheckpointClaimRequest.S
+			if newAssignTo != claimRequest {
```

Comment:
Is there any timeout for claim? what if the stealer just put a claim but never actually claim it?

Reply:
The claim will eventually get stomped on by the record processor checkpointer, then another worker will be eligible to claim it.